private_repo_test.py

```diff
@@ -0,0 +1,183 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2020 ScyllaDB
+
+import os
+import re
+import json
+import time
+
+import requests
+
+from sdcm.tester import ClusterTester
+from sdcm.cluster import get_username
+from sdcm.sct_events import get_logger_event_summary
+from sdcm.utils.common import format_timestamp
+from sdcm.utils.housekeeping import HousekeepingDB
+
+
+REPO_TABLE = "housekeeping.repo"
+REPODOWNLOAD_TABLE = "housekeeping.repodownload"
+PRIVATE_REPO_BASEURL = "https://repositories.scylladb.com/scylla/"
+SW_REPO_RE = re.compile(PRIVATE_REPO_BASEURL + r"repo/(?P<uuid>[\w-]+)/(?P<ostype>[\w]+)")
+
+
+BODY_PREFIXES = {
+    "centos": ("[scylla",
+               "name=",
+               "baseurl=",
+               "type=",
+               "skip_if_unavailable=",
+               "enabled=",
+               "enabled_metadata=",
+               "repo_gpgcheck=",
+               "gpgcheck=",
+               "gpgkey=", ),
+    "ubuntu": ("deb ", ),
+}
+
+
+def extract_pkginfo_urls_from_yum_repo(repofile):
+    urls = []
+    for line in repofile.splitlines():
+        if line.startswith("baseurl="):
+            # Convert string like
+            #   `baseurl=https://.../scylla/centos/scylladb-1.7/$releasever/$basearch/'
+            # to
+            #   `https://.../scylla/centos/scylladb-1.7/7Server/x86_64/repodata/repomd.xml'
+            urls.append(
+                line[8:].replace("$releasever", "7Server").replace("$basearch", "x86_64") + "repodata/repomd.xml")
```

Comment:
From 3.3, the '7Server' directory is removed.   The packages support both CentOS7 & CentOS8.

- http://downloads.scylladb.com/downloads/scylla/rpm/centos/scylladb-3.2/7Server/x86_64/
- http://downloads.scylladb.com/downloads/scylla/rpm/centos/scylladb-3.3/x86_64/

Reply:
Good to know. But it's not a problem for this piece of code: if there is not `$releasever` in the line it'll not be replaced by `7Server`:

```sh
$ curl https://repositories.scylladb.com/scylla/repo/qa-test-centos/centos/scylladb-3.3.repo
[scylla-3.3]
name=Scylla for Centos - $basearch
baseurl=https://repositories.scylladb.com/scylla/downloads/scylladb/qa-test-centos/scylla/rpm/centos/scylladb-3.3/$basearch/
enabled=1
gpgcheck=0

[scylla-generic-3.3]
name=Scylla for centos
baseurl=https://repositories.scylladb.com/scylla/downloads/scylladb/qa-test-centos/scylla/rpm/centos/scylladb-3.3/noarch/
enabled=1
gpgcheck=0

```